src/realm/sync/client_base.hpp

```diff
@@ -62,11 +62,12 @@ class SyncSocketProvider;
 struct ClientReset {
     realm::ClientResyncMode mode;
     DBRef fresh_copy;
-    bool recovery_is_allowed = true;
+    bool recovery_is_allowed;
```

Comment:
Since `SessionErrorInfo` can't be default created and there are only a few parts needed, I don't think we need to. I did remove the `recovery_is_allowed` since it is no longer needed.